src/main/java/com/exactpro/th2/common/schema/message/impl/rabbitmq/AbstractRabbitSubscriber.java

```diff
@@ -68,14 +69,16 @@
                     "The message is meant for any data transferred via RabbitMQ, for example, th2 batch message or event or custom content")
             .register();
 
+    protected final ReentrantLock publicLock = new ReentrantLock();
+    private boolean hasManualSubscriber = false;
+
     protected final String th2Pin;
-    private final List<ConfirmationMessageListener<T>> listeners = new CopyOnWriteArrayList<>();
+    private final CopyOnWriteArrayList<ConfirmationMessageListener<T>> listeners = new CopyOnWriteArrayList<>();
```

Comment:
Do we need CopyOnWriteArrayList if we operate using lock?

Reply:
Yes, the handle operation work with this list without lock. I added lock for public operations only